tmt/steps/provision/connect.py

```diff
@@ -110,8 +117,19 @@ def go(self):
             data['key'] = key
 
         # And finally create the guest
-        self._guest = tmt.Guest(data, name=self.name, parent=self.step)
+        self._guest = GuestConnect(data, name=self.name, parent=self.step)
+
+        if reboot:
+            self._guest.reboot()
 
     def guest(self):
         """ Return the provisioned guest """
         return self._guest
+
+
+class GuestConnect(tmt.Guest):
+    def reboot(self, hard=False):
+        if hard:
+            raise ProvisionErrorUnsupported("Method not supported")
+
+        self.execute("reboot")
```

Comment:
What about providing the soft reboot functionality using `reboot` as the default implementation in the `Guest` class? It represents / can be used for any guest we can ssh to so from the point of view it would make sense to me. Also this part could be inherited / reused by `connect`, `testcloud`, `minute` provision and other plugins such as pure `libvirt` in the future.

I think the responsibility of the method should be also to wait until the guest is back online which is currently misising here completely.

Reply:
If we simply call reboot we should definitely wait for the ssh connection to come back up. It may be possible to reuse some of the logic we've already implemented for machine booting (AFAIK there is somewhere a loop that checks every second for ssh connection until a certain timeout)

Reply:
Not sure about reboot from inside machine.
So proposal is:
wait for ssh connection from reboot() method? And then what? Do we know what we should do after machine reboot?